flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -681,6 +682,13 @@ static constexpr IntrinsicHandler handlers[]{
      {{{"vector_a", asAddr}, {"vector_b", asAddr}}},
      /*isElemental=*/false},
     {"dprod", &I::genDprod},
+    {"eoshift",
+     &I::genEoshift,
+     {{{"array", asAddr},
+       {"shift", asAddr},
+       {"boundary", asValue},
```

Comment:
Per Jean's comments, `boundary` should be pass `asAddr`.

Reply:
Done.  Thanks for the catch!